common/djangoapps/student/roles.py

```diff
@@ -47,9 +47,11 @@ def has_role(self, role, course_id, org):
         """
         Return whether this RoleCache contains a role with the specified role, course_id, and org
         """
+        # Convert course keys to unicode string before comparison. this is because some objects
+        # like ccx locator dont have overridden __eq__. Thats why when we compare them always return false.
         return any(
             access_role.role == role and
-            access_role.course_id == course_id and
+            unicode(access_role.course_id) == unicode(course_id) and
```

Comment:
I don't see a corresponding __eq__ method in https://github.com/edx/opaque-keys/blob/master/opaque_keys/edx/locator.py -- how is this dealt with elsewhere in edX?